openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -121,7 +122,12 @@ func validateAuthenticationSpec(spec configv1.AuthenticationSpec) field.ErrorLis
 				spec.WebhookTokenAuthenticator, fmt.Sprintf("this field cannot be set with the %q .spec.type", spec.Type),
 			))
 		}
+	}
 
+	if spec.Type == configv1.AuthenticationTypeOIDC {
```

Comment:
It might make sense to split the validations of this func into two sub-funcs, one for OAuth and one for OIDC, as it's currently mixed.

Reply:
Are you thinking along the lines of having something like:
```go
switch spec.Type {
    case configv1.AuthenticationTypeOIDC:
        validateOIDC(...)
    case ...:
        ...
}
```
?

I considered doing something like that, but wasn't sold on it being worth the potential scope creep. If you think it would be useful to do that though I can make some changes.

Reply:
I think I would prefer it, but no strong opinion. You could maybe at least separate the respective validations (i.e. put the oidc validation after the oauth metadata) and add some comments to section them out; that'd be fine as well.